tx.go

```diff
@@ -123,6 +127,51 @@ func FromContext(ctx context.Context) (Tx, bool) {
 	return t, ok
 }
 
+type DoneFunc func()
+
+// TraceReq traces an http request. It returns a new context with the transaction
+// included in it, and a func to be called when the request is finished.
+//
+// Usage:
+//
+//     ctx, done := TraceRequest(name, req, ctx)
+//     defer done()
+func TraceRequest(name string, req *http.Request, ctx context.Context) (context.Context, DoneFunc) {
+	tx := NewRequestTx(name, req.URL.String())
+	ctx = WithTx(ctx, tx)
+	tx.Start()
+
+	return ctx, func() {
+		tx.End()
+	}
+}
+
+// TraceExternal adds an external segment to the newrelic transaction, if one exists in the context.
+func TraceExternal(ctx context.Context, host, name string) DoneFunc {
+	return trace(ctx, name, func(tx Tx) {
+		tx.StartExternal(host, name)
```

Comment:
What if these return errors? We may want the DoneFunc to propagate any errors to us. We shouldn't fail, but the calling code may (and probably should) choose to log it.